### PR TITLE
PanelEditor: stores option group collapse state

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
I think this approach is better than #23779. WDYT?

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by dprokop** [comment]
Yes, I like this better!

File: public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const PanelOptionsTab: FC<Props> = ({
   };
   // Fist common panel settings Title, description
   elements.push(
-    <OptionsGroup title="Panel settings" key="Panel settings">
+    <OptionsGroupWithPersistence title="Panel settings" id="Panel settings" key="Panel settings">
```
Comment (dprokop) [inline public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx:51]: Wonder if the persistence shouldn't be controlled via prop. But that's a nit

File: public/app/features/dashboard/components/PanelEditor/OptionsGroupWithPersistence.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  });
+  const onToggle = useCallback(
+    (isExpanded: boolean) => {
+      setValue({ defaultToClosed: !isExpanded });
+      if (props.onToggle) {
+        props.onToggle(isExpanded);
+      }
+    },
+    [setValue, props.onToggle]
+  );
```
Comment (dprokop) [inline public/app/features/dashboard/components/PanelEditor/OptionsGroupWithPersistence.tsx:23]: Would be great to have this as a reusable hook for other collapsible parts of the UI to reuse.

**Comment by hugohaggmark** [comment]
Thanks for quick feedback, then I'll continue replacing all OptionsGroups.

File: public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx:51]: If you had the prop then you could have it set to true by default and not care about replacing OptionsGroup usage

File: public/app/features/dashboard/components/PanelEditor/OptionsGroupWithPersistence.tsx (hunk 2 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/components/PanelEditor/OptionsGroupWithPersistence.tsx:23]: Great suggestion and if I feel I have time for it, I'll do it!

File: public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx:51]: Yea, that was what I was thinking as we’ll

File: public/app/features/dashboard/components/PanelEditor/OptionsGroupWithPersistence.tsx (hunk 2 above)
Comment (dprokop) [reply public/app/features/dashboard/components/PanelEditor/OptionsGroupWithPersistence.tsx:23]: Yeah, that's nice to have if we squeeze transformations and query rows persistance in

File: public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx:51]: I think this is a lot clearer, and I found the OptionsGroup API already so cluttered so I went this path instead.

Comment (torkelo) [reply public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx:51]: For this to be just a built in behavior in the normal OptionsGroup component with option to disable via prop

Comment (dprokop) [reply public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx:51]: I would even go further and refacto OptionsGroup to use something like CollapsibleSection that would hide this behaviour. But maybe that's too much...? :D

Comment (hugohaggmark) [reply public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx:51]: Ok, I'll pass it as a prop then So 2 new props on OptionsGroup then `id` and `persistMe`?

Comment (dprokop) [reply public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx:51]: ThumbsUp™

**Comment by torkelo** [comment]
Looking good!

**Review by torkelo — APPROVED** [review]

**Comment by dprokop** [comment]
<3

**Pull request merged**